velox/exec/Exchange.cpp

```diff
@@ -58,20 +60,17 @@ void SerializedPage::prepareStreamForDeserialize(ByteStream* input) {
 std::shared_ptr<ExchangeSource> ExchangeSource::create(
     const std::string& taskId,
     int destination,
-    std::shared_ptr<ExchangeQueue> queue) {
+    std::shared_ptr<ExchangeQueue> queue,
+    memory::MemoryPool* pool) {
   for (auto& factory : factories()) {
-    auto result = factory(taskId, destination, queue);
+    auto result = factory(taskId, destination, queue, pool);
```

Comment:
Let's add VELOX_CHECK_NOT_NULL(pool) at the beginning of this method.

Reply:
Shall we just do FOLLY_NONNULL from the input?